go/vt/discovery/fake_healthcheck.go

```diff
@@ -55,9 +51,9 @@ func NewFakeHealthCheck(ch chan *TabletHealth) *FakeHealthCheck {
 // FakeHealthCheck implements discovery.HealthCheck.
 type FakeHealthCheck struct {
 	// mu protects the items map
-	mu    sync.RWMutex
-	items map[string]*fhcItem
-
+	mu               sync.RWMutex
+	items            map[string]*fhcItem
+	currentTabletUID sync2.AtomicInt32
```

Comment:
nit: couldn't we use a normal int and use the mutex to protect it?

Reply:
it is not a bad idea, wherever we use `currentTabletUID` (in `Reset` & `AddFakeTablet`) we acquire the mutex's lock, I'll change that to an `int`

Reply:
fixed via 4b16187ea7